code/game/objects/items/devices/flashlight.dm

```diff
@@ -293,62 +294,139 @@
+		fuel = rand(25 MINUTES, 35 MINUTES)
+
+/obj/item/flashlight/flare/toggle_light()
+	if(on || !fuel)
+		return FALSE
+
+	name = "lit [initial(name)]"
+	attack_verb_continuous = string_list(list("burns", "singes"))
+	attack_verb_simple = string_list(list("burn", "singe"))
+	hitsound = 'sound/items/welder.ogg'
+	. = ..()
+
+/obj/item/flashlight/flare/proc/turn_off()
+	on = FALSE
+	name = initial(name)
+	attack_verb_continuous = initial(attack_verb_continuous)
+	attack_verb_simple = initial(attack_verb_simple)
+	hitsound = initial(hitsound)
+	force = initial(force)
+	damtype = initial(damtype)
+	update_brightness()
+
+/obj/item/flashlight/flare/extinguish()
+	if(fuel != INFINITY)
+		turn_off()
+	return ..()
+
+/obj/item/flashlight/flare/update_brightness()
+	..()
+	inhand_icon_state = "[initial(inhand_icon_state)]" + (on ? "-on" : "")
+	update_appearance()
 
 /obj/item/flashlight/flare/process(delta_time)
 	open_flame(heat)
-	fuel = max(fuel -= delta_time, 0)
-	if(fuel <= 0 || !on)
+	fuel = max(fuel -= delta_time * (1 SECONDS), 0)
+
+	if(!fuel || !on)
 		turn_off()
-		if(!fuel)
-			icon_state = "[initial(icon_state)]-empty"
 		STOP_PROCESSING(SSobj, src)
 
+		if(!fuel && trash_type)
+			new trash_type(loc)
+			qdel(src)
+
 /obj/item/flashlight/flare/ignition_effect(atom/A, mob/user)
-	. = fuel && on ? span_notice("[user] lights [A] with [src] like a real badass.") : ""
+	if(get_temperature())
+		. = span_notice("[user] lights [A] with [src].")
 
-/obj/item/flashlight/flare/proc/turn_off()
-	on = FALSE
-	force = initial(src.force)
-	damtype = initial(src.damtype)
-	if(ismob(loc))
-		var/mob/U = loc
-		update_brightness(U)
-	else
-		update_brightness(null)
+/obj/item/flashlight/flare/proc/ignition(mob/user)
+	if(user && !fuel)
+		to_chat(user, span_warning("[src] is out of fuel!"))
+		return FALSE
+	if(user && on)
+		to_chat(user, span_warning("[src] is already lit!"))
+		return FALSE
+	if(!toggle_light())
+		return FALSE
 
-/obj/item/flashlight/flare/update_brightness(mob/user = null)
-	..()
-	if(on)
-		inhand_icon_state = "[initial(inhand_icon_state)]-on"
-	else
-		inhand_icon_state = "[initial(inhand_icon_state)]"
+	force = on_damage
+	damtype = BURN
+	if(fuel != INFINITY)
+		START_PROCESSING(SSobj, src)
 
-/obj/item/flashlight/flare/attack_self(mob/user)
-	// Usual checks
-	if(fuel <= 0)
-		to_chat(user, span_warning("[src] is out of fuel!"))
-		return
-	if(on)
-		to_chat(user, span_warning("[src] is already on!"))
-		return
+	return TRUE
 
-	. = ..()
-	// All good, turn it on.
-	if(.)
+/obj/item/flashlight/flare/fire_act(exposed_temperature, exposed_volume)
+	ignition()
+	return ..()
+
+/obj/item/flashlight/flare/attack_self(mob/user)
+	if(ignition(user))
 		user.visible_message(span_notice("[user] lights \the [src]."), span_notice("You light \the [src]!"))
-		force = on_damage
-		damtype = BURN
-		START_PROCESSING(SSobj, src)
 
 /obj/item/flashlight/flare/get_temperature()
 	return on * heat
 
+/obj/item/flashlight/flare/candle
```

Comment:
this subtype feels very evil, maybe /obj/item/flashlight/flare should be generalized to something like /obj/item/flashlight/fueled

Reply:
I already considered adding a `/obj/item/flashlight/fuel` subtype but ultimately decided it wasn't necessary.  We don't have enough items with deviating properties to make it worth it.